### PR TITLE
Update thermal eigenstrain API again 

### PR DESCRIPTION
**Pull request by dschwen**
Refs #295

This still requires a MOOSE submodule update.

### CODE REVIEW

File: src/materials/NEMLThermalExpansionEigenstrain.C [hunk 1]
```diff
@@ -40,20 +40,17 @@ NEMLThermalExpansionEigenstrain::NEMLThermalExpansionEigenstrain(const InputPara
...
+NEMLThermalExpansionEigenstrain::computeThermalStrain()
 {
-  double nemlCTE = _model->alpha(_temperature[_qp]);
+  double nemlCTE = _model->alpha(MetaPhysicL::raw_value(_temperature[_qp]));
   double nemlCTE_old = _model->alpha(_temperature_old[_qp]);
 
-  thermal_strain =
+  const auto thermal_strain =
       _tstrain_old[_qp] + (nemlCTE + nemlCTE_old) / 2 * (_temperature[_qp] - _temperature_old[_qp]);
+  _tstrain[_qp] = MetaPhysicL::raw_value(thermal_strain);
```
Comment (bwspenc) [inline src/materials/NEMLThermalExpansionEigenstrain.C:51]: ...another instance of the _thermal_strain we were just talking about!

**Review by bwspenc — COMMENTED** [review]
Looks good. We'll need a submodule update here too of course.

File: src/materials/NEMLThermalExpansionEigenstrain.C (hunk 1 above)
Comment (dschwen) [reply src/materials/NEMLThermalExpansionEigenstrain.C:51]: Yeah, I'll update that with the new `dual_number_cast`

Comment (dschwen) [reply src/materials/NEMLThermalExpansionEigenstrain.C:51]: Isn't T-strain a rapper?

**Review by bwspenc — APPROVED** [review]

**Pull request merged**